### PR TITLE
kube-proxy: OnServiceUpdate []*api.Service

### PR DESCRIPTION
**Pull request by thockin**
This signature is more consistent with OnEndpointsUpdate and removes a
copy loop.  This is part on ongoing cleanup to rate-limit iptables
calls.

### CODE REVIEW

**Comment by wojtek-t** [comment]
/lgtm

@thockin - I'm assuming that you checked this is safe and nothing id modifying services in proxiers. When I was changing it for endpoints I checked that for endpoints, thought didn't yet get to checking it for services. If that's the case, feel free to remove the do-not-merge label.

**Comment by thockin** [comment]
Sorry, I am accumulating a bunch of related commits before sending them all out for review.  I looked for any places that wrote through that list and didn't see any (and they would be broken anyway, on updates?).

**Comment by wojtek-t** [comment]
> Sorry, I am accumulating a bunch of related commits before sending them all out for review. I looked for any places that wrote through that list and didn't see any (and they would be broken anyway, on updates?).

Yeah - that hopefully should trigger some test failures.

So can you let me know when this will be ready for review? Also - can we merge this in smaller parts (I think that this PR is already big enough for merging)?

File: pkg/proxy/config/config.go [hunk 1]
```diff
@@ -33,9 +33,17 @@ import (
...
 // ServiceConfigHandler is an abstract interface of objects which receive update notifications for the set of services.
 type ServiceConfigHandler interface {
-	// OnServiceUpdate gets called when a configuration has been changed by one of the sources.
-	// This is the union of all the configuration sources.
-	OnServiceUpdate(services []api.Service)
+	// OnServiceUpdate gets called when a service is created, removed or changed
+	// on any of the configuration sources. An example is when a new service
+	// comes up.
+	//
+	// NOTE: For efficiency, endpoints are being passed by reference, thus,
```
Comment (ncdc) [inline pkg/proxy/config/config.go]: s/endpoints/services/?

**Comment by thockin** [comment]
OK, I dialed it back to just the one commit.  I'll open a new one (or several) for the rest

**Comment by wojtek-t** [comment]
/lgtm

Thanks!

**Pull request merged**